src/github.com/couchbase/sync_gateway/db/kv_change_index_writer.go

```diff
@@ -313,60 +326,48 @@ func (k *kvChangeIndexWriter) indexEntries(entries []*LogEntry, indexPartitions
 
 		// Track vbucket sequences for clock update
 		updatedSequences.SetSequence(logEntry.VbNo, logEntry.Sequence)
-
 	}
 
 	// Wait group tracks when the current buffer has been completely processed
 	var channelWg sync.WaitGroup
-	channelStart := time.Now()
+
+	channelErrorCount := uint32(0)
 	// Iterate over channel sets to update channel index
 	for channelName, entrySet := range channelSets {
 		channelWg.Add(1)
-		go func(channelName string, entrySet []*LogEntry) {
+		go func(channelName string, entrySet []*LogEntry, errorCount uint32) {
 			defer channelWg.Done()
-			k.addSetToChannelIndex(channelName, entrySet)
-
-		}(channelName, entrySet)
+			err := k.addSetToChannelIndex(channelName, entrySet)
+			if err != nil {
+				atomic.AddUint32(&errorCount, 1)
+			}
+		}(channelName, entrySet, channelErrorCount)
 	}
 
 	// Wait for entry and channel processing to complete
 	entryWg.Wait()
 	channelWg.Wait()
+	if atomic.LoadUint32(&entryErrorCount) > 0 || atomic.LoadUint32(&channelErrorCount) > 0 {
+		return errors.New("Unrecoverable error indexing entry or channel")
+	}
 
-	indexTimingExpvars.Add("batch_writeEntriesAndChannelBlocks", time.Since(batchStart).Nanoseconds())
-	indexTimingExpvars.Add("batch_writeChannelBlocks", time.Since(channelStart).Nanoseconds())
-
-	writeHistogram(indexTimingExpvars, batchStart, "indexEntries-entryAndChannelTime")
-
-	stableStart := time.Now()
 	// Update stable sequence
 	err := k.getWriterStableSequence().UpdateAndWrite(updatedSequences)
-	if err != nil {
-		base.LogPanic("Error updating stable sequence", err)
-	}
-
-	indexTimingExpvars.Add("batch_updateStableSequence", time.Since(stableStart).Nanoseconds())
-	writeHistogram(indexTimingExpvars, stableStart, "indexEntries-stableSeqTime")
-
-	// TODO: remove - iterate once more for perf logging
-	for _, logEntry := range entries {
-		writeHistogram(indexTimingExpvars, logEntry.TimeReceived, "lag-indexing")
-		writeHistogram(indexTimingExpvars, logEntry.TimeSaved, "lag-totalWrite")
-	}
-
-	writeHistogram(indexTimingExpvars, batchStart, "indexEntries-batchTime")
-	indexTimingExpvars.Add("batch_totalBatchTime", time.Since(batchStart).Nanoseconds())
+	return err
 }
 
-func (k *kvChangeIndexWriter) addSetToChannelIndex(channelName string, entries []*LogEntry) {
+func (k *kvChangeIndexWriter) addSetToChannelIndex(channelName string, entries []*LogEntry) error {
 	writer, err := k.getOrCreateWriter(channelName)
 	if err != nil {
 		base.LogFatal("Unable to obtain channel writer - partition map not defined?")
+		return err
```

Comment:
this return will never get invoked since it's calling `base.LogFatal()` above.. is that what was intended?